perspective.el

```diff
@@ -1043,34 +1052,33 @@ perspective beginning with the given letter."
 
 (defun persp--state-window-state-massage (entry persp valid-buffers)
-  "This is a primitive code walker. It removes references to
-potentially problematic buffers from the data structure created
-by window-state-get and replaces them with references to the
-perspective-specific *scratch* buffer. Buffers are considered
-'problematic' when they have no underlying file, or are otherwise
-transient.
-
-The need for a recursive walk, and the consequent complexity of
-this function, arises from the nature of the data structure
-returned by window-state-get. That data structure is essentially
-a tree represented as a Lisp list. It can contain several kinds
-of nodes, including properties, nested trees representing window
-splits, and windows (referred to internally as leaf nodes).
-
-For the purposes of preserving window state, we only care about
-nodes in this data structure which refer to buffers, i.e., lists
-with the symbol 'buffer in the first element. These 'buffer lists
-can be deeply buried inside the data structure, because it
-recursively describes the layout of all windows in the given
-frame. They are always nested in lists with the symbol 'leaf in
-the first element.
+  "This is a primitive code walker.
+It remove references to potentially problematic buffers from the data
+structure created by `window-state-get and replace them with
+references to the perspective-specific *scratch* buffer.  Buffers are
+considered 'problematic' when they have no underlying file, or are
+otherwise transient.
+
+The need for a recursive walk, and the consequent complexity of this
+function, arises from the nature of the data structure returned by
+`window-state-get'.  That data structure is essentially a tree
+represented as a Lisp list.  It can contain several kinds of nodes,
+including properties, nested trees representing window splits, and
+windows (referred to internally as leaf nodes).
+
+For the purposes of preserving window state, we only care about nodes
+in this data structure which refer to buffers, i.e., lists with the
+symbol 'buffer in the first element.  These 'buffer lists can be
+deeply buried inside the data structure, because it recursively
+describes the layout of all windows in the given frame.  They are
+always nested in lists with the symbol 'leaf in the first element.
 
 And so, the walker descends the data structure and preserves
-everything it finds. When it notices a 'leaf, it iterates over
-its properties until it finds a 'buffer. If the 'buffer points to
-a buffer which can be reasonably saved, it leaves it alone.
-Otherwise, it replaces that buffer's node with one which points
-to the perspective's *scratch* buffer."
+everything it finds.  When it notices a 'leaf, it iterates over its
+properties until it finds a 'buffer.  If the 'buffer points to a buffer
+which can be reasonably saved, it leaves it alone.  Otherwise, it
+replaces that buffer's node with one which points to the perspective's
+*scratch* buffer."
```

Comment:
idgi

Reply:
This diff is just `M-q` and lisp conventions. Double-space is needed after a period.

Reply:
@conao3: Thank you for looking into this! I very much appreciate the effort you put into improving the docstrings, especially where you added clarifications about parameter behavior.

That said, can we please _not_ use two spaces after a period in perspective source code? The convention you're referring to has not been recommended for over 25 years in the Chicago Manual of Style (https://www.chicagomanualofstyle.org/qanda/data/faq/topics/OneSpaceorTwo/faq0007.html) or the Modern Language Association (https://style.mla.org/number-of-spaces-after-period/), the two most authoritative bodies in American English writing. On a personal note, I have always found two spaces profoundly difficult to look at in monospaced typefaces. It leads to a jagged look, and makes the problem of rivers in text worse (https://en.wikipedia.org/wiki/River_(typography)). `(setq sentence-end-double-space nil)` is one of the earliest entries I put in my Emacs configuration. I know it breaks sentence movement commands, but the tradeoff is worth it to me.

Reply:
Long-time no see @gcv! Since this issue ( https://github.com/tumashu/posframe/issues/29 ).
It is very good that an active maintainer can manage this package.
This seems to be different agreed upon, and in this case, the maintainer's opinion is respected.
Close PR.

Reply:
I didn't mean for you to close the whole PR, and apologize if my comment was too harsh. I merged in most of your suggestions in commit https://github.com/nex3/perspective-el/commit/599aa7df7805c507cf06eeb1756d6388f1b3b3fd.

Reply:
No need to apologize! And it is good you become a collaborator!